hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/TssLibraryImpl.java

```diff
@@ -0,0 +1,152 @@
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.tss;
+
+import static com.hedera.node.app.tss.handlers.TssUtils.SIGNATURE_SCHEMA;
+
+import com.hedera.cryptography.bls.BlsPrivateKey;
+import com.hedera.cryptography.bls.BlsPublicKey;
+import com.hedera.cryptography.bls.BlsSignature;
+import com.hedera.cryptography.tss.api.TssMessage;
+import com.hedera.cryptography.tss.api.TssParticipantDirectory;
+import com.hedera.cryptography.tss.api.TssParticipantPrivateInfo;
+import com.hedera.cryptography.tss.api.TssPrivateShare;
+import com.hedera.cryptography.tss.api.TssPublicShare;
+import com.hedera.cryptography.tss.api.TssService;
+import com.hedera.cryptography.tss.api.TssShareSignature;
+import com.hedera.cryptography.tss.impl.Groth21Service;
+import com.hedera.node.app.spi.AppContext;
+import com.hedera.node.app.tss.api.TssLibrary;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.state.lifecycle.info.NodeInfo;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.security.SecureRandom;
+import java.util.List;
+import java.util.function.Supplier;
+import javax.inject.Inject;
+
+public class TssLibraryImpl implements TssLibrary {
+    private static final BlsPrivateKey AGGREGATED_PRIVATE_KEY =
+            BlsPrivateKey.create(SIGNATURE_SCHEMA, new SecureRandom());
+    private byte[] message = new byte[0];
+
+    private TssService tssService;
+    private SecureRandom random = new SecureRandom();
+    private Supplier<NodeInfo> selfNodeInfo;
+
+    @Inject
+    public TssLibraryImpl(AppContext appContext) {
+        this.tssService = new Groth21Service(SIGNATURE_SCHEMA, random);
+        this.selfNodeInfo = appContext.selfNodeInfoSupplier();
+    }
+
+    @NonNull
+    @Override
+    public TssMessage generateTssMessage(@NonNull final TssParticipantDirectory tssParticipantDirectory) {
+        return tssService.genesisStage().generateTssMessage(tssParticipantDirectory);
+    }
+
+    @NonNull
+    @Override
+    public TssMessage generateTssMessage(
+            @NonNull TssParticipantDirectory tssParticipantDirectory, @NonNull TssPrivateShare privateShare) {
+        return tssService.rekeyStage().generateTssMessage(tssParticipantDirectory, privateShare);
+    }
+
+    @Override
+    public boolean verifyTssMessage(
+            @NonNull TssParticipantDirectory participantDirectory, @NonNull Bytes tssMessageBytes) {
+        try {
+            final var tssMessage = tssService.messageFromBytes(participantDirectory, tssMessageBytes.toByteArray());
+            return tssService.rekeyStage().verifyTssMessage(participantDirectory, null, tssMessage);
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    @NonNull
+    @Override
+    public List<TssPrivateShare> decryptPrivateShares(
+            @NonNull TssParticipantDirectory participantDirectory, @NonNull List<TssMessage> validTssMessages) {
+        // TODO: How to get the BlsPrivateKey here
```

Comment:
I see. Thank you! I think we will have the private key. But this will be fixed after the `tssEncryptionKey` proposal is implemented.

Reply:
This is all confusing because everything is a key. 
But we are speaking about two different keys:

The key you mention is the key the participant uses to decrypt the shares other participants sent to it in tss messages. That one will never be aggregated, that's right, you'll start by having that.

The key I am speaking about is the ledger id private key, similarly to what happens with public shares, if you do the same for private shares, you get a private key.  But if somebody recovers that key, then that somebody can sign message in name of the whole network, which of course we don't want. But all is safe considering that nodes only have share number of private shares, unless they can get threshold number of them, there is no problem, and that is why this operation is not necessary.

Reply:
Ah got it. Thanks for clarification. Removed the comment. This must use correct key in next PRs.